tasks/main.yml

```diff
@@ -67,8 +52,30 @@
 - name: load config from file
   ansible.builtin.shell: "{{ ldaptoolbox_openldap_slapd_cli_cmd }} restoreconfig"
 
+- name: build generated data from template
+  ansible.builtin.shell: "{{ ldaptoolbox_openldap_slapd_cli_cmd }} importdatatemplate"
+  when: ldaptoolbox_openldap_generate_data_ldif is defined
+
 # Import Data
 # -----------
 
-#TODO
+- name: import data ldif provided
+  copy:
+    src: "{{ ldaptoolbox_openldap_import_data_ldif }}"
+    dest: "{{ ldaptoolbox_openldap_configuration_backup_dir }}/{{ ldaptoolbox_openldap_data_prefix }}-{{ ldaptoolbox_openldap_configuration_timestamp }}.ldif"
+    owner: "{{ ldaptoolbox_openldap_configuration_owner }}"
+    group: "{{ ldaptoolbox_openldap_configuration_group }}"
+    mode: "{{ ldaptoolbox_openldap_configuration_mode }}"
+  when: ldaptoolbox_openldap_import_data_ldif is defined
+
+- name: add syncrepl account
+  include_tasks: ldaptoolbox-add-syncrepl.yml
+  when: ldaptoolbox_openldap_import_data_ldif is defined
+
+- name: add default ppolicy pwdPolicy
+  include_tasks: ldaptoolbox-add-ppolicy.yml
+  when: ldaptoolbox_openldap_import_data_ldif is defined
 
+- name: restore data
+  ansible.builtin.shell: "{{ ldaptoolbox_openldap_slapd_cli_cmd }} restore"
+  when: ( ldaptoolbox_openldap_import_data_ldif is defined ) or ( ldaptoolbox_openldap_generate_data_ldif is defined )
```

Comment:
restoring data is useless and will overwrite the current data when doing a `ldaptoolbox_openldap_generate_data_ldif` task